py-polars/tests/unit/test_selectors.py

```diff
@@ -814,3 +814,11 @@ def test_selector_list_of_lists_18499() -> None:
 
     with pytest.raises(InvalidOperationError, match="invalid selector expression"):
         lf.unique(subset=[["bar", "ham"]])  # type: ignore[list-item]
+
+
+def test_invalid_selector() -> None:
+    df = pl.DataFrame(data={"x": [1, 2], "z": ["a", "b"]})
+    with pytest.raises(InvalidOperationError, match="invalid selector expression"):
+        df.drop(pl.col("x", "z") + 2)  # type: ignore[arg-type]
```

Comment:
Whys does mypy complain here?

Reply:
```
❯ mypy py-polars/tests/unit/test_selectors.py
py-polars/tests/unit/test_selectors.py:822: error: Argument 1 to "drop" of "DataFrame" has incompatible type "Expr"; expected "str | _selector_proxy_ | Iterable[str | _selector_proxy_]"  [arg-type]
py-polars/tests/unit/test_selectors.py:824: error: Argument 1 to "drop" of "DataFrame" has incompatible type "Expr"; expected "str | _selector_proxy_ | Iterable[str | _selector_proxy_]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```

Reply:
Then I think we must pass a `seelctor` as we don't allow expression inputs here.

Reply:
i could reproduce the issue as well with the following:

```py
pl.selectors.all() + pl.col("x")
```

which is at runtime a `_selector_proxy_`, mypy still thinks it's a `Expr`, hence the `assert` in [`3947452` (#19136)](https://github.com/pola-rs/polars/pull/19136/commits/3947452bc1061030e34cc59a81de830213a162d1)